### PR TITLE
Move smart-contract-verifier

### PR DESCRIPTION
**Pull request by sevenzing**

### CODE REVIEW

**Review by leviska — APPROVED** [review]

File: smart-contract-verifier/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+COPY --from=cache $CARGO_HOME $CARGO_HOME
+RUN cargo build --release
+
+FROM ubuntu:20.04 as run
+RUN apt-get update && apt-get install -y libssl1.1 libssl-dev ca-certificates
+
+WORKDIR /app
+ENV APP_USER=app
+RUN groupadd $APP_USER \
+    && useradd -g $APP_USER $APP_USER
```
Comment (rimrakhimov) [inline smart-contract-verifier/Dockerfile:25]: Could you please add comments, why those lines are required

File: smart-contract-verifier/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+COPY --from=build /app/target/release/smart-contract-verifier-http /app/smart-contract-verifier-http
+RUN chown -R $APP_USER:$APP_USER /app
```
Comment (rimrakhimov) [inline smart-contract-verifier/Dockerfile:28]: Could you please add comment, why this line is required

File: smart-contract-verifier/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+RUN cargo chef cook --release --recipe-path recipe.json
+
+FROM chef AS build
+
+COPY . .
+COPY --from=cache /app/target target
+COPY --from=cache $CARGO_HOME $CARGO_HOME
+RUN cargo build --release
+
+FROM ubuntu:20.04 as run
```
Comment (rimrakhimov) [inline smart-contract-verifier/Dockerfile:19]: Nit:

In cargo-chef description they use slim version of debian system in their example:
```
# We do not need the Rust toolchain to run the binary!
FROM debian:buster-slim AS runtime
```
Maybe we can also reuse that approach to decrease the size of our docker image?

**Review by rimrakhimov — COMMENTED** [review]
Could you please extract docker-compose file into the external `smart-contract-verifier` directory and update Readme to be in line with current installation and new links (at least [Building from sources](https://github.com/blockscout/blockscout-rs/tree/lymarenkolev/move-scv/smart-contract-verifier/smart-contract-verifier-http#building-from-source) section)

File: smart-contract-verifier/Dockerfile (hunk 3 above)
Comment (sevenzing) [reply smart-contract-verifier/Dockerfile:19]: I tried to use `debian:buster-slim`, `distroless` and also several "small-weight" images, but vyper compiler always return error `lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.29' not found`. 

`debian:bullseye-slim` image worked, but it had the same image size as ubuntu, so I decided to leave ubuntu

**Review by rimrakhimov — APPROVED** [review]

**Pull request merged**